crypto/mem_sec.c

```diff
@@ -433,8 +438,19 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(OPENSSL_SYS_LINUX) && defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
+    if (syscall(SYS_mlock2, sh.arena, sh.arena_size, MLOCK_ONFAULT) < 0) {
+        if (errno == ENOSYS) {
+            if (mlock(sh.arena, sh.arena_size) < 0)
+                ret = 2;
+        } else {
```

Comment:
It's kind of a trick question. It was my original reaction too, but judging from equivalence between "before" and "after", it's correct. Because success is "claimed" by *not* modifying 'ret'. One can wonder if this is a good way to "claim" success, but that how it's done here. [Though I don't actually defend the choice, only saying that original behaviour is preserved.]